src/test/java/Homework23.java

```diff
@@ -0,0 +1,16 @@
+import org.testng.annotations.Test;
+
+public class Homework23 extends BaseTest{
+    @Test(dataProvider = "IncorrectLoginProviders")
+    public void LoginTest (String email, String password) throws InterruptedException {
+        openLoginUrl();
+        LoginPage login= new LoginPage(driver);
+        login.provideEmail("demo@class.com");
+        login.providePassword("te$t$tudent");
+        login.clickSubmit();
+
```

Comment:
Please, add Assert

**One test - One Assert**

For example

![Clean_Test](https://github.com/Test-Pro-Bootcamp/A47-QA-Automation-Framework/assets/104761482/20f89bdf-936f-4df1-b9a4-1f3fcdc139c9)